src/Selenium2Driver.php

```diff
@@ -707,7 +708,24 @@ public function setValue($xpath, $value)
   document.activeElement.blur();
 }
 JS;
-        $this->executeJsOnElement($element, $script);
+
+        // In some very rare edge cases, a developer might have implemented some
+        // functionality whereby entering a specific value could cause the field
+        // to be detached from the DOM.
+        //
+        // e.g. a 2FA input field that immediately recognizes the correct value
+        // as it is typed, and causes the SPA to update the page instantly.
+        //
+        // If that happens, we will get a StaleElementReference exception when
+        // we try to execute JS on it. As such, we catch that condition here and
+        // ignore it since an element is obviously no longer focused if it is
+        // no longer attached to the page document.
```

Comment:
Please replace with:

```php
// Cover case, when an element was removed from DOM after its value was
// changed (e.g. by a JavaScript of a SPA) and therefore can't be focused.
```

---

People always can use git annotate to locate associated PR, which explains all in much greater detail.